tests/glsl-intrinsic/intrinsic-functional.slang

```diff
@@ -0,0 +1,2319 @@
+//TEST(compute):COMPARE_COMPUTE_EX(filecheck-buffer=DX11):-allow-glsl -compute -dx11 -output-using-type
+//TEST(compute):COMPARE_COMPUTE_EX(filecheck-buffer=DX12):-allow-glsl -compute -dx12 -use-dxil -output-using-type -profile sm_6_0
+//TEST(compute, vulkan):COMPARE_COMPUTE_EX(filecheck-buffer=VK):-allow-glsl -compute -vk -emit-spirv-directly -output-using-type -profile sm_6_0 -render-feature hardware-device
+
+#define TRIG_FUNCTIONS 0
+#define EXP_FUNCTIONS 1
+#define COMMON_FUNCTIONS 2
+#define PACK_FUNCTIONS 3
+#define GEOMETRIC_FUNCTIONS 4
+#define MATRIX_FUNCTIONS 5
+#define VEC_FUNCTIONS 6
+#define INT_FUNCTIONS 7
```

Comment:
I think these could be better with enum

Reply:
The problem with enum with these is in the main function I'd have to do this:
outputBuffer[(uint)IntrinsicType::Trigonometry] = AngleAndTrig();

Which is a lot less clean than:
outputBuffer[EXP_FUNCTIONS] = Exponential();

Reply:
Can you use `[UnscopedEnum]` that is [described in our doc](https://github.com/shader-slang/slang/blob/4b3f554a58e4224806c31d66874fbe60f1f09332/docs/user-guide/02-conventional-features.md)?